File: UniMate/app/src/main/java/com/example/umpsa/MainActivity.java
package com.example.umpsa;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;

public class MainActivity extends Activity {

    View cardAcademic;
    View cardAiLearning;
    View cardCommunity;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_home);

        NavigationHelper.setup(this, "home");

        cardAcademic = findViewById(R.id.cardAcademic);
        cardAiLearning = findViewById(R.id.cardAiLearning);
        cardCommunity = findViewById(R.id.cardCommunity);

        cardAcademic.setOnClickListener(v ->
                startActivity(new Intent(MainActivity.this, ScheduleActivity.class))
        );

        cardAiLearning.setOnClickListener(v ->
                startActivity(new Intent(MainActivity.this, AiLearningActivity.class))
        );

        cardCommunity.setOnClickListener(v ->
                startActivity(new Intent(MainActivity.this, CommunityActivity.class))
        );
    }
}

File: manidk/app/src/main/java/com/example/manidk/MainActivity.kt
package com.example.manidk

import android.app.Activity
import android.os.Bundle
import android.widget.Button
import android.widget.ViewFlipper
import androidx.core.graphics.toColorInt

class MainActivity : Activity() { // <-- We changed this to standard Activity

    private lateinit var viewFlipper: ViewFlipper
    private lateinit var btnNavReport: Button
    private lateinit var btnNavManage: Button

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        // Hide standard Action Bar for a clean, modern look
        actionBar?.hide() // <-- We changed this to match the new Activity type

        setContentView(R.layout.activity_home)

        // Map UI Elements
        viewFlipper = findViewById(R.id.viewFlipper)
        btnNavReport = findViewById(R.id.btnNavReport)
        btnNavManage = findViewById(R.id.btnNavManage)

        // Bottom Navigation Listeners
        btnNavReport.setOnClickListener {
            viewFlipper.displayedChild = 0 // Show Reports Page
            updateNavState(true)
        }

        btnNavManage.setOnClickListener {
            viewFlipper.displayedChild = 1 // Show Manage Page
            updateNavState(false)
        }
    }

    // Handles the color switching of the Bottom Navigation to show active states
    private fun updateNavState(isReportActive: Boolean) {
        if (isReportActive) {
            btnNavReport.setTextColor("#7C3AED".toColorInt()) // Active Purple
            btnNavManage.setTextColor("#6B7280".toColorInt()) // Inactive Gray
        } else {
            btnNavReport.setTextColor("#6B7280".toColorInt()) // Inactive Gray
            btnNavManage.setTextColor("#7C3AED".toColorInt()) // Active Purple
        }
    }
}